caas/kubernetes/provider/k8s_test.go

```diff
@@ -49,35 +51,52 @@ type K8sSuite struct {
 
 var _ = gc.Suite(&K8sSuite{})
 
-func (s *K8sSuite) TestMakeUnitSpecNoConfigConfig(c *gc.C) {
-	podSpec := caas.PodSpec{
-		ProviderPod: &provider.K8sPodSpec{
+func float64Ptr(f float64) *float64 {
+	return &f
+}
+
+func int32Ptr(i int32) *int32 {
+	return &i
+}
+
+func int64Ptr(i int64) *int64 {
+	return &i
+}
+
+func boolPtr(b bool) *bool {
+	return &b
+}
+
+func (s *K8sSuite) TestMakeSvcSpecNoConfigConfig(c *gc.C) {
+	podSpec := specs.PodSpec{
+		ServiceAccount: &specs.ServiceAccountSpec{
+			Name:                         "serviceAccount",
+			AutomountServiceAccountToken: boolPtr(true),
+		},
+	}
+
+	podSpec.ProviderPod = &k8sspecs.K8sPodSpec{
+		Pod: &k8sspecs.PodSpec{
+			RestartPolicy:                 core.RestartPolicyOnFailure,
 			ActiveDeadlineSeconds:         int64Ptr(10),
-			ServiceAccountName:            "serviceAccount",
-			Hostname:                      "host",
-			Subdomain:                     "sub",
-			DNSPolicy:                     core.DNSClusterFirst,
```

Comment:
Still need DNSPolicy, Priority, PriorityClassName as k8s specific attributes
I see DNSPolicy and Priority below, but not PriorityClassName?